src/circuit/value_commit_orchard.rs

```diff
@@ -0,0 +1,391 @@
+        // Evaluate commitment = [v_net_magnitude_sign] asset
+        let commitment = match zsa_params {
+            None => {
+                let v_net = ScalarFixedShort::new(
+                    ecc_chip.clone(),
+                    layouter.namespace(|| "v_net"),
+                    v_net_magnitude_sign,
+                )?;
+
+                // commitment = [v_net] ValueCommitV
+                let (commitment, _) = {
+                    let value_commit_v = ValueCommitV;
+                    let value_commit_v =
+                        FixedPointShort::from_inner(ecc_chip.clone(), value_commit_v);
+                    value_commit_v.mul(layouter.namespace(|| "[v] ValueCommitV"), v_net)?
+                };
+                commitment
+            }
+            Some(params) => {
+                // Check that magnitude is 64 bits
+                // Note: if zsa_params is not provided, this check will be performed inside the
+                // fixed-base short scalar multiplication.
+                {
+                    let lookup_config = params.sinsemilla_chip.config().lookup_config();
+                    let (magnitude_words, magnitude_extra_bits) = (6, 4);
+                    assert_eq!(
+                        magnitude_words * sinsemilla::primitives::K + magnitude_extra_bits,
+                        64
+                    );
+                    let magnitude_zs = lookup_config.copy_check(
+                        layouter.namespace(|| "magnitude lowest 60 bits"),
+                        v_net_magnitude_sign.0.clone(),
+                        magnitude_words, // 6 windows of 10 bits.
+                        false,           // Do not constrain the result here.
+                    )?;
+                    assert_eq!(magnitude_zs.len(), magnitude_words + 1);
+                    lookup_config.copy_short_check(
+                        layouter.namespace(|| "magnitude highest 4 bits"),
+                        magnitude_zs[magnitude_words].clone(),
+                        magnitude_extra_bits, // The 7th window must be a 4 bits value.
+                    )?;
+                }
+
+                // Multiply asset by magnitude, using the long scalar mul.
+                // TODO: implement a new variable base multiplication which is optimized for 64-bit scalar
+                // (the long scalar mul is optimized for pallas::Base scalar (~255-bits))
+                //
+                // magnitude_asset = [magnitude] asset
+                let magnitude_asset = {
+                    let magnitude_scalar = ScalarVar::from_base(
+                        ecc_chip.clone(),
+                        layouter.namespace(|| "magnitude"),
+                        &v_net_magnitude_sign.0,
+                    )?;
+                    let (magnitude_asset, _) = params
+                        .asset
+                        .mul(layouter.namespace(|| "[magnitude] asset"), magnitude_scalar)?;
```

Comment:
_From str4d ([r2599387594](https://github.com/zcash/orchard/pull/471#discussion_r2599387594))_

Rename this field to be clearer:
```suggestion
                        .asset_base
                        .mul(layouter.namespace(|| "[magnitude] asset_base"), magnitude_scalar)?;
```

Reply:
CIRCUIT